payment_payfip/models/payment_provider.py

```diff
@@ -0,0 +1,373 @@
+        mel = response.find("mel")
+        montant = response.find("montant")
+        numcli = response.find("numcli")
+        objet = response.find("objet")
+        refdet = response.find("refdet")
+        saisie = response.find("saisie")
+
+        data = {
+            "resultrans": resultrans.text if resultrans is not None else False,
+            "dattrans": dattrans.text if dattrans is not None else False,
+            "heurtrans": heurtrans.text if heurtrans is not None else False,
+            "exer": exer.text if exer is not None else False,
+            "idOp": idOp.text if idOp is not None else False,
+            "mel": mel.text if mel is not None else False,
+            "montant": montant.text if montant is not None else False,
+            "numcli": numcli.text if numcli is not None else False,
+            "objet": objet.text if objet is not None else False,
+            "refdet": refdet.text if refdet is not None else False,
+            "saisie": saisie.text if saisie is not None else False,
+        }
+
+        return data
+
+    def _payfip_check_web_service(self):
+        self.ensure_one()
+
+        error = _("It would appear that the customer number entered is not valid or that the PayFIP contract is " "not properly configured.")
+
+        soap_url = self._get_soap_url()
+        soap_body = """
+                    <soapenv:Envelope %s %s>
+                       <soapenv:Header/>
+                       <soapenv:Body>
+                          <pai:recupererDetailClient>
+                             <arg0>
+                                <numCli>%s</numCli>
+                             </arg0>
+                          </pai:recupererDetailClient>
+                       </soapenv:Body>
+                    </soapenv:Envelope>
+                    """ % (
+            'xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/"',
+            'xmlns:pai="http://securite.service.tpa.cp.finances.gouv.fr/services/mas_securite/' 'contrat_paiement_securise/PaiementSecuriseService"',
+            self.payfip_customer_number,
+        )
+
+        try:
+            soap_response = requests.post(soap_url, data=soap_body, headers={"content-type": "text/xml"})
+        except ConnectionError:
+            _logger.error(error)
+            return False, error
+
+        root = ElementTree.fromstring(soap_response.content)
+        fault = root.find(".//S:Fault", {"S": "http://schemas.xmlsoap.org/soap/envelope/"})
+
+        if fault is not None:
+            error_desc = fault.find(".//descriptif")
+            if error_desc is not None:
+                error += _('\nPayFIP server returned the following error: "%s"') % error_desc.text
+            _logger.error(error)
+            _logger.debug(fault)
+            return False, error
+
+        return True, ""
+
+    def toggle_payfip_activation_mode_value(self):
+        in_activation = self.filtered(lambda acquirer: acquirer.payfip_activation_mode)
```

Comment:
```suggestion
        in_activation = self.filtered('payfip_activation_mode')
```
Tips : pas besoin de lambda si on évalue juste la présence du champ, les deux fonctionnent donc cela est à votre convenance